### PR TITLE
[Apps Framework] Separate calls

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add separate calls for each function on Apps (submit, form source, form submit and lookup). Also updated the filtering with the new logic.

This PR also piggybacks the renaming of markdown and error to text, and filtering out "white space labels" (to avoid no text labels on post options and channel header), and some improvements on the command parser.

Notes for testing: This is a breaking change. All Apps will need to be updated to work properly. The test app will be updated to fit the new approach.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39092
https://mattermost.atlassian.net/browse/MM-39096
https://mattermost.atlassian.net/browse/MM-31121
https://mattermost.atlassian.net/browse/MM-33532

#### Related Pull Requests
Apps Plugin: https://github.com/mattermost/mattermost-plugin-apps/pull/296
Webapp: https://github.com/mattermost/mattermost-webapp/pull/9263
Test App: TBD

#### Release Note
```release-note
Breaking change for Apps: calls are now separated between submit, form, refresh and lookup calls.
```

### CODE REVIEW

**Review by levb — CHANGES_REQUESTED** [review]
Great work, just one question about the order of checking in handleBindingClick

File: app/actions/apps.ts [hunk 1]
```diff
@@ -3,17 +3,76 @@
...
 import {CommandArgs} from '@mm-redux/types/integrations';
 import {Post} from '@mm-redux/types/posts';
-import {cleanForm, makeCallErrorResponse} from '@utils/apps';
+import {cleanForm, createCallRequest, makeCallErrorResponse} from '@utils/apps';
 
-export function doAppCall<Res=unknown>(call: AppCallRequest, type: AppCallType, intl: any): ActionFunc {
+export function handleBindingClick<Res=unknown>(binding: AppBinding, context: AppContext, intl: any): ActionFunc {
+    return async (dispatch: DispatchFunc) => {
+        // Fetch form
+        if (binding.form?.source) {
```
[the remark this answers is not in the document]
Comment (larkox) [reply app/actions/apps.ts:16]: A binding has either a submit, or a form. Cannot have both (will be filtered out). That solves the precedence problem on that side.
Now the precedence between a form with only the source, and a form with source and content, we could switch them, but I personally think that it is better to fetch the form if it has a source.

File: app/screens/apps_form/apps_form_container.tsx [hunk 2]
```diff
@@ -183,54 +185,33 @@ export default class AppsFormContainer extends PureComponent<Props, State> {
...
-            },
-            values: {
-                ...call.values,
-            },
-        };
+        const creq = createCallRequest(field.lookup, this.props.context, {}, {
+            ...values,
+            [field.name]: userInput,
+        });
+        return this.props.actions.doAppLookup(creq, intl);
```
Comment (mickmister) [inline app/screens/apps_form/apps_form_container.tsx:203]: @levb If I understand this correctly, `query: userInput` has changed to `[field.name]: userInput`.

I find this confusing, as `field.type` is a select element in this case, but its value is being "submitted" as a query string for the lookup. The type mismatch seems a little off.

What if we want to have a multiselect lookup field, and have the app check the currently selected values during the lookup? It seems there's no way to access that here.

Comment (levb) [reply app/screens/apps_form/apps_form_container.tsx]: Discussed offline, decided to reinstate `CallRequest.LookupQuery` along with `LookupField`

**Comment by mickmister** [comment]
@DHaussermann

Tested with the Test App, the ServiceNow App, and the Zendesk App

### Test App

Went through all:

- Channel header bindings
- Post menu bindings
- Command bindings
- Embedded forms

All functions in these areas seem to be working correctly, having the same behavior as on the webapp.

###  Service Now

- Authentication works
- Creating a ticket works

### Zendesk

- Authentication works
- Creating a ticket works
- Filling out a subscription form works

Note that since Zendesk targets are deprecated, I needed to fake out the code into thinking there was a valid target available. As a result, I was not able to properly test the submission of the subscriptions form.

### Other info

There is a known issue with select fields not being pre-selected with default values. This is fixed with https://github.com/mattermost/mattermost-mobile/pull/5640

**Pull request merged**